wisdem/glue_code/glue_code.py

```diff
@@ -213,107 +176,151 @@ def setup(self):
             self.connect("configuration.turb_class", "rotorse.rp.gust.turbulence_class")
 
-            # Connections to RotorStructure
-            self.connect("blade.internal_structure_2d_fem.d_f", "rotorse.rs.brs.d_f")
-            self.connect("blade.internal_structure_2d_fem.sigma_max", "rotorse.rs.brs.sigma_max")
-            self.connect("blade.pa.chord_param", "rotorse.rs.brs.rootD", src_indices=[0])
-            self.connect("blade.ps.layer_thickness_param", "rotorse.rs.brs.layer_thickness")
-            self.connect("blade.internal_structure_2d_fem.layer_start_nd", "rotorse.rs.brs.layer_start_nd")
-            self.connect("blade.internal_structure_2d_fem.layer_end_nd", "rotorse.rs.brs.layer_end_nd")
-            if modeling_options["WISDEM"]["RotorSE"]["bjs"]:
-                self.connect("materials.name", "rotorse.rs.bjs.name_mat")
-                self.connect("materials.rho", "rotorse.rs.bjs.rho_mat")
-                self.connect("materials.Xt", "rotorse.rs.bjs.Xt_mat")
-                self.connect("materials.sigma_y", "rotorse.rs.bjs.Xy_mat")
-                self.connect("materials.E", "rotorse.rs.bjs.E_mat")
-                self.connect("materials.S", "rotorse.rs.bjs.S_mat")
-                self.connect("blade.pa.chord_param", "rotorse.rs.bjs.chord")
-                self.connect("blade.high_level_blade_props.r_blade", "rotorse.rs.bjs.blade_length")
-                self.connect("blade.ps.layer_thickness_param", "rotorse.rs.bjs.layer_thickness")
-                self.connect("blade.internal_structure_2d_fem.layer_width", "rotorse.rs.bjs.layer_width")
-                self.connect("blade.internal_structure_2d_fem.layer_start_nd", "rotorse.rs.bjs.layer_start_nd")
-                self.connect("blade.internal_structure_2d_fem.layer_end_nd", "rotorse.rs.bjs.layer_end_nd")
-                self.connect("blade.interp_airfoils.coord_xy_interp", "rotorse.rs.bjs.coord_xy_interp")
-                self.connect("blade.interp_airfoils.r_thick_interp", "rotorse.rs.bjs.rthick")
-                self.connect("blade.internal_structure_2d_fem.joint_position", "rotorse.rs.bjs.joint_position")
-                self.connect(
-                    "blade.internal_structure_2d_fem.joint_nonmaterial_cost", "rotorse.rs.bjs.joint_nonmaterial_cost"
-                )
-                self.connect(
-                    "blade.internal_structure_2d_fem.reinforcement_layer_ss", "rotorse.rs.bjs.reinforcement_layer_ss"
-                )
+            if modeling_options["WISDEM"]["RotorSE"]["inn_af"]:
+                self.connect("blade.run_inn_af.coord_xy_interp", "rotorse.re.coord_xy_interp")
+            elif not modeling_options["WISDEM"]["RotorSE"]["user_elastic"]:
+                self.connect("blade.interp_airfoils.coord_xy_interp", "rotorse.re.coord_xy_interp")
+
+            # Connections to rotor elastic and frequency analysis
+            if not modeling_options["WISDEM"]["RotorSE"]["user_elastic"]:
```

Comment:
I might be confused here, but I don't see the connections for the else condition...

Reply:
Oh I see now, you just assign them in `gc_WT_InitModel.py`. that probably works, but it's not consistent with the rest of the code where a first set of outputs is first initialized and then propagated to all submodules. @gbarter any thought?

Reply:
@ptrbortolotti - I'm not tracking the alternative that you are proposing.  Maybe discuss?

Reply:
@yqliaohk and I touched base yesterday and she was going to make some edits!

Reply:
@gbarter Working on this! The direction I am heading towards is to move the independent var component to `gc_WT_DataStruc.py`, connect them to components in rotorse in `glue_code.py`, and initialize them in `gc_WT_InitModel.py` instead of creating the ivc directly in rotorse and skipping `gc_WT_DataStruc.py` and connections in `glue_code.py`.